contracts/eth/nearprover/utils/upgrade_admin.js

```diff
@@ -0,0 +1,67 @@
+const { ethers } = require('hardhat');
+const { assert, expect } = require('chai');
+const { EthLedgerSigner } = require('./eth-ledger-signer');
+
+async function getSlotsData (provider, contractAddress, numOfSlotsToDisplay) {
+    for (let i = 0; i < numOfSlotsToDisplay; ++i) {
+        const slotData = await provider.getStorageAt(contractAddress, i);
+        console.log(slotData);
+    }
+}
+
+async function upgradeAdminAddressTo ({
+    provider,
+    contractAddress,
+    currentAdminAddress,
+    newAdminAddress,
+    adminAddressSlot,
+    ledgerKeyPath,
+}) {
+    const currentAdmin = ethers.BigNumber.from(await provider.getStorageAt(contractAddress, Number(adminAddressSlot))).toHexString();
+    assert.equal(
+        currentAdmin,
+        currentAdminAddress,
+        `The current admin doesn't match at the slot ${adminAddressSlot} contract ${contractAddress}`,
+    );
+
+    const adminControlledFactory = await ethers.getContractFactory('AdminControlled');
+    const adminControlled = adminControlledFactory.attach(contractAddress);
+
+    console.log(`Current admin: ${await adminControlled.admin()}`);
+    console.log(`Trying to upgrade admin address to: ${newAdminAddress}`);
+
+    let signer;
+    // We use non-strict unequality as it also includes undefined, etc
+    if (ledgerKeyPath != null) {
+        signer = new EthLedgerSigner(provider, ledgerKeyPath);
+    } else {
+        signer = new ethers.Wallet(process.env.ETH_PRIVATE_KEY, provider);
+    }
+
+    assert.equal(
+        await signer.getAddress(),
+        await adminControlled.admin(),
+        `The used account is not an admin of contract ${contractAddress}`,
+    );
+
+    // Mask matches only on the latest 20 bytes (to store the address)
+    const mask = ethers.BigNumber.from('0x000000000000000000000000ffffffffffffffffffffffffffffffffffffffff');
+    const response = await adminControlled
+        .connect(signer)
+        .adminSstoreWithMask(adminAddressSlot, newAdminAddress, mask);
```

Comment:
As this is quite critical, after logging all that information, let's create some kind of prompt asking to type something like "confirm" or "approve" before actually calling the command. 
If someone uses Ledger they still have some time to think before approving the operation, but for the regular calls, it will be executed immediately even if incorrect slot/data was provided. We need to handle this.

Reply:
Fixed at aaca1ea9a057b92cce7331f25250eb55b7b83b52